pkg/manager/internal.go

```diff
@@ -150,6 +154,45 @@ type controllerManager struct {
 	retryPeriod time.Duration
 }
 
+type errSignaler struct {
+	// errSignal indicates that an error occurred, when closed.  It shouldn't
+	// be written to.
+	errSignal chan struct{}
```

Comment:
Why it's `chan struct{}` instead of `chan error`?
`nil` can be treated as `error`, right?

Reply:
Since this is being used purely for signaling that some error has occurred, it is fine.